elide-core/src/test/java/com/yahoo/elide/core/security/obfuscation/AesBytesEncryptor.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2024, the original author or authors.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.security.obfuscation;
+
+import java.security.spec.KeySpec;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+public class AesBytesEncryptor {
+    private final SecretKey secretKey;
+
+    public AesBytesEncryptor(SecretKey secretKey) {
+        this.secretKey = secretKey;
+    }
+
+    public AesBytesEncryptor(String password, String salt) {
+        this(from(password, salt));
+    }
+
+    static SecretKey from(String password, String salt) {
+        try {
+            SecretKeyFactory factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA256");
+            KeySpec spec = new PBEKeySpec(password.toCharArray(), salt.getBytes(), 65536, 256);
+            return new SecretKeySpec(factory.generateSecret(spec).getEncoded(), "AES");
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    byte[] encrypt(byte[] byteArray) {
+        try {
+            byte[] key = secretKey.getEncoded();
+            SecretKeySpec secretKeySpec = new SecretKeySpec(key, "AES");
```

Comment:
Does this encryption include randomization or padding?  The ID should be stable so we should hash instead of encrypt.

Reply:
This class is just for testing purposes. I felt it was best that users used a dedicated crypto library eg. https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/crypto/encrypt/BytesEncryptor.html which would provide more options and this approach is less likely to introduce a CVE into the Elide library.

The FunctionIdObfuscator should make it easy to integrate with other libraries.